src/features/pass-survey/model/AnswerValidator.ts

```diff
@@ -164,12 +177,12 @@ export function AnswerValidator(
     },
 
     isEqualToRowOption(rowIndex: number, optionValue: string) {
-      const selectedOption = getRowOptionValues(rowIndex);
+      const selectedOption = getSelectionRowOptionValue(rowIndex);
       return selectedOption !== null && selectedOption.includes(optionValue);
     },
 
     isNotEqualToRowOption(rowIndex: number, optionValue: string) {
-      const selectedOption = getRowOptionValues(rowIndex);
+      const selectedOption = getSelectionRowOptionValue(rowIndex);
```

Comment:
Ran into an issue here, if the value is null it doesn't pass this validation (`value != option`, which `null !== option`)

* Single Select Item configured as a 2-item matrix

![CleanShot 2024-12-03 at 21 11 11](https://github.com/user-attachments/assets/212f586e-ceef-4095-a174-5c1e85fc8420)

* Condition says:
  * if first row = o1, show message1
  * if second row != o1, show message2

![CleanShot 2024-12-03 at 21 09 58](https://github.com/user-attachments/assets/dadb7ce1-9e4b-486f-a9be-91ce4707710d)

Video:
Second row is null and it fails validation, even though its allowed by the item type

https://github.com/user-attachments/assets/b0f047b5-6dc6-4c59-bb1e-accb8718a180

Reply:
It  looks right for me, for example, if we enable the null as a right answer, it means every condition will pass as long you choose only one option.

Should I considerer null as valid answer ?

Reply:
in my understanding the logic is like: if the row 2 of item "single is selected turn it visible, otherwise keep it invisible, turn null into right answer, we are saying: "ok if dont select we are considerer you want to see it"

Reply:
its looks right for me, because we are not accepting null as a right answer.

Example, I understand the logic is like: ok if you select something different of row2 - o1 it will appears, but if we consider empty or null as a right value, we are saying: "ok, even if you dont select nothing, we are considering your answer as different of conditional option"

Reply:
let me know your thoughts, should we enable null as different in terms of conditional logic comparison  ?

Reply:
if yes, it is fast to solve.